pyOCD/coresight/cortex_m.py

```diff
@@ -876,11 +665,15 @@ def readCoreRegistersRaw(self, reg_list):
         reg_vals = []
         for reg, reg_cb, dhcsr_cb in zip(reg_list, reg_cb_list, dhcsr_cb_list):
             dhcsr_val = dhcsr_cb()
-            assert dhcsr_val & CortexM.S_REGRDY
+#            assert dhcsr_val & CortexM.S_REGRDY
             val = reg_cb()
 
-            # Special handling for registers that are combined
-            # into a single DCRSR number.
+            # If we couldn't read the register, stuff a dummy zero in the return value list.
+            if (dhcsr_val & CortexM.S_REGRDY) == 0:
+                reg_vals.append(0)
```

Comment:
Instead of adding a dummy value here, could an exception be raised?  The calling code could then catch the appropriate exception if they want to be able to recover from this.